### DISCUSSION TITLE
How to connect the websocket mechanism and how to check that it is enabled from the outside?

### DISCUSSION THREAD

**Discussion by marisarze** [filecoin-project/lotus#12786] [Developers/Service Providers] [answered]
Hi. When starting a lotus node, how do I enable the websocket mechanism? What is the default address for websocket? Do I understand these addresses correctly?
http url: http://x.x.x.x:1234/rpc/v1
websocket url: ws://x.x.x.x:1234/rpc/v0

http url works well and connects. But websocket does not work.
I set flags:

```
[Fevm]
  # EnableEthRPC enables eth_ RPC methods.
  # Note: Setting this to true will also require that ChainIndexer is enabled, otherwise it will cause an error at startup.
  #
  # type: bool
  # env var: LOTUS_FEVM_ENABLEETHRPC
  EnableEthRPC = true

[API]
  # Binding address for the Lotus API
  #
  # type: string
  # env var: LOTUS_API_LISTENADDRESS
  ListenAddress = "/ip4/0.0.0.0/tcp/1234/http"

  # type: string
  # env var: LOTUS_API_REMOTELISTENADDRESS
  RemoteListenAddress = "x.x.x.x:1234"

  # type: Duration
  # env var: LOTUS_API_TIMEOUT
  Timeout = "30s" 
```
are there any other flags that need to be changed
how to check that websocket is working fine from external "api" call?

**Comment by rvagg** [comment, marked as answer]
Websockets is enabled on the same port and URL as the HTTP endpoint - websockets is just an "upgrade" for an HTTP connection so the same port serves that purpose. If `http://x.x.x.x:1234/rpc/v1` works, then `ws://x.x.x.x:1234/rpc/v1` will work, unless you have something odd going on with your network or websockets library.

You can use a library like `wscat` to debug if you want, e.g.:

```
$ wscat -c ws://localhost:1234/rpc/v1 -x '{"method":"Filecoin.ChainHead","params":[],"id":2,"jsonrpc":"2.0"}'
{"id":2,"jsonrpc":"2.0","result":{"Cids":[{"/":"bafy2bzacebkij3dit4cvz7xhnxv76ggbbanigfapte5y3xdmwap74gfkp7m42"},{"/":"bafy2bzacea5apv66shtoxg6wdt36wt2u2wqlz74h3s2p3bvr6lrfbu5klau7w"},{"/":"bafy2bzaceaso3edv56jbk5goqcfi3hpagdapxqlwms4jfsvamplkwfhdcp7b4"},{"/":"bafy2bzaceca4yqee7pv4qsbvvvlktkr7kqk256toln4ecp3igzg2s5vvoonvq"},
...
```

**Reply by marisarze** [reply]
Oh thank you, I feel better) I'm satisfied with everything.
